File: remove-duplicates-linkedlist/RemoveDuplicateLinkedList.swift
import Foundation

public class RemoveDuplicateLinkedList {
    class Node {
        public var next: Node?
        public var value: Int
        public init(value: Int) {
            self.value = value
            self.next = nil
        }
    }

    func removeDuplcatesLinkedList(_ linkedList: Node) -> Node {
        var current: Node? = linkedList
        while current != nil {
            var inner = current!.next
            while inner != nil, current!.value == inner!.value {
                inner = inner!.next
            }
            current!.next = inner
            current = inner
        }
        return linkedList
    }
}

File: remove-duplicates-linkedlist/RemoveDuplicateLinkedList.cs
using System.Collections.Generic;
using System;


public class RemoveDuplicateLinkedList {
	public class Node {
		public int value;
		public Node next;

		public Node(int value) {
			this.value = value;
			this.next = null;
		}
	}

    // 1 -> 1 -> 1 -> 4 -> 4 -> 4 -> 5 -> 6 -> 6 -> null
	public Node RemoveDuplicatesFromLinkedList(Node linkedList) {
        Node current = linkedList;
        while(current != null) {
            Node inner = current.next;
            while(inner != null && current.value == inner.value) {
                inner = inner.next;
            }
            current.next = inner;
            current = inner;
        }
        return linkedList;
    }
}